crates/ide-db/src/rename.rs

```diff
@@ -180,15 +180,35 @@ fn rename_mod(
     let InFile { file_id, value: def_source } = module.definition_source(sema.db);
     if let ModuleSource::SourceFile(..) = def_source {
         let anchor = file_id.original_file(sema.db);
-        // not mod.rs and doesn't has children, rename file only
-        if !module.is_mod_rs(sema.db) && module.children(sema.db).next().is_none() {
+
+        let is_mod_rs = module.is_mod_rs(sema.db);
+        let has_detached_child = module
+            .children(sema.db)
+            .any(|child| child.definition_source(sema.db).file_id != file_id);
```

Comment:
This parses all those modules (not a big issue in this case really but its better to avoid parsing where possible), so it would be nice to avoid that. Could you add a function to `hir::Module` like `is_outline`, and similarly an `is_outline` function on `ModuleOrigin` to call, based on https://github.com/rust-lang/rust-analyzer/blob/master/crates/hir-def/src/nameres.rs#L171-L176

Reply:
Thanks for pointing it out. Couldn't figure it out myself that I'm literally checking if a child not inline. However, I felt that adding another function just for this case would be overkill (and I couldn't find any usage of the term "outline" in this context), so I just rewrote it with `!child.is_inline(sema.db)`. Ping me if you still think otherwise and I'll rewrite it via `is_outline` function.